lib/internal/event_target.js

```diff
@@ -559,20 +560,25 @@ function emitUnhandledRejectionOrErr(that, err, event) {
 
 function defineEventHandler(emitter, name) {
   // 8.1.5.1 Event handlers - basically `on[eventName]` attributes
-  let eventHandlerValue;
   ObjectDefineProperty(emitter, `on${name}`, {
     get() {
-      return eventHandlerValue;
+      return this[kHandlers]?.get(name);
     },
     set(value) {
-      if (eventHandlerValue) {
-        emitter.removeEventListener(name, eventHandlerValue);
+      const oldValue = this[kHandlers]?.get(name);
+      if (oldValue) {
+        this.removeEventListener(name, oldValue);
       }
       if (typeof value === 'function') {
-        emitter.addEventListener(name, value);
+        this.addEventListener(name, value);
       }
-      eventHandlerValue = value;
-    }
+      if (!this[kHandlers]) {
+        this[kHandlers] = new Map();
```

Comment:
```suggestion
        this[kHandlers] = new SafeMap();
```

Reply:
Good catch!

Reply:
It looks like `kEvents` also needs to be a SafeMap since it does not currently use primordial methods to access its properties.

Reply:
updated, ptal